crates/fj-proc/src/lib.rs

```diff
@@ -91,13 +95,67 @@ pub fn model(_: TokenStream, input: TokenStream) -> TokenStream {
 
     match parse::parse(&item) {
         Ok(init) => {
-            let item = without_param_attrs(item);
+            let mut item = without_param_attrs(item);
+
+            // Yes, all of this is to add `fj::abi::initialize_panic_handling();` to the top of the function.
+            item.block.stmts.insert(
+                0,
+                Stmt::Semi(
+                    Expr::Call(ExprCall {
+                        attrs: vec![],
+                        func: Box::new(Expr::Path(ExprPath {
+                            attrs: vec![],
+                            qself: None,
+                            path: syn::Path {
+                                leading_colon: None,
+                                segments: {
+                                    let mut segments = Punctuated::new();
+
+                                    segments.push(PathSegment {
+                                        ident: Ident::new(
+                                            "fj",
+                                            Span::call_site(),
+                                        ),
+                                        arguments: PathArguments::None,
+                                    });
+
+                                    segments.push(PathSegment {
+                                        ident: Ident::new(
+                                            "abi",
+                                            Span::call_site(),
+                                        ),
+                                        arguments: PathArguments::None,
+                                    });
+
+                                    segments.push(PathSegment {
+                                        ident: Ident::new(
+                                            "initialize_panic_handling",
+                                            Span::call_site(),
+                                        ),
+                                        arguments: PathArguments::None,
+                                    });
+
+                                    segments
+                                },
+                            },
+                        })),
+                        paren_token: Paren {
+                            span: Span::call_site(),
+                        },
+                        args: Punctuated::new(),
+                    }),
+                    syn::token::Semi::default(),
+                ),
+            );
```

Comment:
I'm far from an expert in procedural macros and might be wrong about what's happening here, but couldn't this be done much easier using the `quote!` macro? See [`expand.rs`](https://github.com/hannobraun/Fornjot/blob/main/crates/fj-proc/src/expand.rs) for examples.